### DISCUSSION TITLE
getServerSideProps inside React Class

### DISCUSSION THREAD

**Discussion by buddies2705** [vercel/next.js#12281] [Help]
I am trying to covert my CRA app to SSR using Nextjs.

I want to know how to use **`getServerSideProps use in react class component`** 

**What I am using?**
 - MUI
 - React Bootstrap
 - Express

**What do I want?**

I have a class where I am fetching data and setting in react state, class is written based on Client-side fetching. Now I added in the `page` directory, and want to load the initial state, so the page can be rendered on the server when the user fetches it the first time. 

But my fetching functions and login around is inside the class, so if I add `getServerSideProps` then how will I call a method which inside the class. 

**Note**: I am a noob.

**Comment by rafaelalmeidatk** [comment]
`getServerSideProps` is a named export, so it is defined the same way if your class is a function or a class: https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering

```js
export async function getServerSideProps(context) {
  return {
    props: {}, // will be passed to the page component as props
  }
}
```

You don't call class methods from `getServerSideProps`, it is a function that will fetch some data and send to your component through props

**Reply by Ekeu** [reply]
Just as a side note this code snippet is in my `_app.js` file

**Reply by rafaelalmeidatk** [reply]
`getServerSideProps` is not available in `_app`, only `getInitialProps`. Check the caveats: https://nextjs.org/docs/advanced-features/custom-app#caveats

**Reply by Ekeu** [reply]
Ah yeah, reason why I deleted my question :laughing: . I just went through the docs once more.

Thanks for your feedback :+1:
